test-app/apps/backend-v2/src/app/shift/shift.dto.ts

```diff
@@ -1,4 +1,21 @@
 import { IsInt, IsOptional, IsString, Length, Min } from "class-validator";
+import { createZodDto } from "nestjs-zod";
+import { z } from "nestjs-zod/z";
+
+import { datePreprocessSchema, laterDateAllowEqual } from "../utils";
+
+const BaseShift = z.object({
+  start: datePreprocessSchema(),
+  end: datePreprocessSchema(),
+  maxVolunteers: z.number().min(1),
+  title: z.string().min(1),
+  description: z.union([z.string().min(1), z.undefined()]),
+  eventAdmin: z.string(),
+});
```

Comment:
Is this not in the shared package? Hoping we can avoid redefining things.

Reply:
Yep updated the shared package implementation to be compatible here